libsolidity/interface/CompilerStack.cpp

```diff
@@ -626,6 +650,43 @@ CompilerStack::Source const& CompilerStack::source(string const& _sourceName) co
 	return it->second;
 }
 
+string CompilerStack::createOnChainMetadata(Contract const& _contract) const
+{
+	Json::Value meta;
+	meta["version"] = 1;
+	meta["language"] = "Solidity";
+	meta["compiler"]["version"] = VersionString;
+	meta["compilationTarget"] =
+		_contract.contract->sourceUnitName() +
+		":" +
+		_contract.contract->annotation().canonicalName;
+
+	for (auto const& s: m_sources)
+	{
+		solAssert(s.second.scanner, "Scanner not available");
+		meta["sources"][s.first]["swarm"] =
```

Comment:
We can include that in the spec, but I do not really see the full point of that, at least as long as the swarm hash is still extremely similar to the sha3 hash.